specs/Slots.md

```diff
@@ -0,0 +1,64 @@
+# Slots API specification
+
+Fluent UI components almost always contain sub parts and these pre-defined layouts can be configured via Slots API. Slots are named areas in a component that can receive props and provide styling and layout for them.
+
+## Slots definition
+
+For example, `Button` component contains `content`, `loader` and `icon` slots in its layout:
+
+```jsx
+<root>
+  <loader />
+  <content />
+  <icon />
+</root>
+```
+
+## Slots usage
+
+Slots can be configured through props, which lets the caller pass in a variety of inputs for a given slot. There are several forms of values that can be provided, but all of them share one common thing - they will produce a proper JSX layout inside a component.
+
+### An object as a value
+
+If you will pass an object to component props we will handle it as props for a slot. In an example below, props will be passed to an `icon` slot:
+
+```jsx
+<Button icon={{ children: <FooIcon />, id: '#button-icon' }} />
```

Comment:
**q:** I'm bit confused about the API TBH. 

This example will render following?

```diff
<root>
  <loader />
  <content />
+  icon slot render as -> <FooIcon id="button-icon" />
</root>
```

or like this?

```diff
<root>
  <loader />
  <content />
+ <icon id="button-icon">
+     <FooIcon />
+</icon>
</root>
```

Reply:
I'm not 100% sure but I think it will be the former. @layershifter am I correct here?

Reply:
Actually, I think it's always (or by default) within a `span` element. See [react-utils/src/compose/getSlots.ts#L33](https://github.com/microsoft/fluentui/blob/master/packages/react-utils/src/compose/getSlots.ts#L33).

Reply:
@kubkon @Hotell `id` will go to a slot while `icon` will be rendered inside. I agree that this without an output it's confusing, so I added a snippet with rendered HTML. Is it cleaner now?